pkg/config/remote/service/service.go

```diff
@@ -907,6 +911,29 @@ func filterNeededTargetFiles(neededConfigs []string, cachedTargetFiles []*pbgo.T
 	return filteredList, nil
 }
 
+func (s *CoreAgentService) apiKeyUpdateCallback(dbPath string) func(string, any, any) {
+	return func(setting string, _, newvalue any) {
+		s.Lock()
+		defer s.Unlock()
+
+		if setting != "api_key" {
+			return
+		}
+
+		newKey, ok := newvalue.(string)
+		if ok {
+			s.api.UpdateAPIKey(newKey)
+		}
+		apiKeyHash := hashAPIKey(newKey)
+		db, err := recreate(dbPath, s.agentVersion, apiKeyHash)
```

Comment:
At one point we were talking about only updating the bucket inside the boltDB that tracks the metadata such as agent version and API key so we don't have to destroy the database when this happens. Was something discovered that makes that not possible? I'm concerned a bit about the implications here of destroying the entire TUF repo live out from underneath actively initialized TUF clients. It may work in local testing but there's a lot of different permutations of scenarios that would be difficult to validate exhaustively.